pkg/store/gcworker/gc_worker_test.go

```diff
@@ -777,6 +778,59 @@ func TestDeleteRangesFailure(t *testing.T) {
 	}
 }
 
+func TestConcurrentDeleteRanges(t *testing.T) {
+	// make sure the parallelization of deleteRanges works
+
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/store/gcworker/mockHistoryJobForGC", "return(1)"))
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/pkg/store/gcworker/mockHistoryJob", "return(\"schema/d1/t1\")"))
+	defer func() {
+		require.NoError(t, failpoint.Disable("github.com/pingcap/tidb/pkg/store/gcworker/mockHistoryJobForGC"))
+		require.NoError(t, failpoint.Disable("github.com/pingcap/tidb/pkg/store/gcworker/mockHistoryJob"))
+	}()
+
+	s := createGCWorkerSuite(t)
+	se := createSession(s.gcWorker.store)
+	defer se.Close()
+	_, err := se.Execute(gcContext(), `INSERT INTO mysql.gc_delete_range VALUES
+("1", "2", "31", "32", "10"),
+("3", "4", "33", "34", "10"),
+("5", "6", "35", "36", "15"),
+("7", "8", "37", "38", "15"),
+("9", "10", "39", "40", "15")
+	`)
+	require.NoError(t, err)
+
+	ranges, err := util.LoadDeleteRanges(gcContext(), se, 20)
+	require.NoError(t, err)
+	require.Len(t, ranges, 5)
+
+	stores, err := s.gcWorker.getStoresForGC(context.Background())
+	require.NoError(t, err)
+	require.Len(t, stores, 3)
+	sort.Slice(stores, func(i, j int) bool { return stores[i].Address < stores[j].Address })
+
+	sendReqCh := make(chan SentReq, 20)
+	s.client.unsafeDestroyRangeHandler = func(addr string, req *tikvrpc.Request) (*tikvrpc.Response, error) {
+		sendReqCh <- SentReq{req, addr}
+		resp := &tikvrpc.Response{
+			Resp: &kvrpcpb.UnsafeDestroyRangeResponse{},
+		}
+		return resp, nil
+	}
+	defer func() { s.client.unsafeDestroyRangeHandler = nil }()
+
+	err = s.gcWorker.deleteRanges(gcContext(), 20, gcConcurrency{3, false})
+	require.NoError(t, err)
+
+	s.checkDestroyRangeReq(t, sendReqCh, ranges, stores)
+
```

Comment:
Perhaps we could also check the ranges are deleted in 2 rounds with max-concurrency  set to `3`.

Reply:
There's no need for the concept of 'rounds' here. Consider a scenario with 3 workers completing 3, 1, and 1 ranges respectively. This would also be a valid result.